src/timing.rs

```diff
@@ -93,6 +94,20 @@ impl Timeout {
     }
 }
 
+/// With exponential backoff, repeatedly try the callback until the result is `Ok`
+pub fn retry_until_ok<T, E, F: FnMut() -> Result<T, E>>(base: u64, mut callback: F) -> T {
```

Comment:
There should be a maximum amount of time (it is not desirable to have a backoff time go to hours or days).  For example, to from 1s to a maximum of 5m by multiplying by 2 each iteration; and making maximum configurable.